src/linker/Linker/Annotations.cs

```diff
@@ -244,7 +245,7 @@ public bool IsInstantiated (TypeDefinition type)
 			return marked_instantiated.Contains (type);
 		}
 
-		public void MarkRelevantToVariantCasting (TypeDefinition type)
+		public void MarkRelevantToVariantCasting (TypeDefinition? type)
```

Comment:
I'd prefer to make the type non-nullable, and require callers to decide what to do if for example a type doesn't resolve.